org.eclipse.scanning.server/src/org/eclipse/scanning/server/servlet/ScanProcess.java

```diff
@@ -387,20 +387,20 @@ private IDeviceController createRunnableDevice(ScanBean bean, IPointGenerator<?>
 	}
 
 	private void configureDetectors(Map<String, Object> dmodels, ScanModel model, IPointGenerator<?> generator) throws Exception {
-		
+
 		logger.debug("Configuring detectors {}", dmodels!=null?dmodels.keySet():null);
 		for (IRunnableDevice<?> device : model.getDetectors()) {
-			
+
 			AnnotationManager manager = new AnnotationManager(Activator.createResolver());
 			manager.addDevices(device);
 			manager.addContext(model.getScanInformation());
-			
+
 			@SuppressWarnings("unchecked")
 			IRunnableDevice<Object> odevice = (IRunnableDevice<Object>)device;
-			
+
 			if (!dmodels.containsKey(odevice.getName())) continue; // Nothing to configure
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "dmodels" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)